src/layouts/components/Header.tsx

```diff
@@ -227,10 +227,10 @@ function HeaderNetworkMenu (): JSX.Element {
   )
 }
 
-function HeaderLink (props: { text: string, pathname: string, className: string }): JSX.Element {
+function HeaderLink (props: { text: string, pathname: string, className: string, route?: string }): JSX.Element {
   return (
```

Comment:
Would it be better to use `const router = useRouter()` here instead of passing it in? You keep the props inheritance tree clean and use a more natural react construct to get deeply nested information. This change also reduces the complexity of spiderwebbing the code.

Reply:
That is right